### PR TITLE
Fix: Resolve issue where case-sensitive column pruning removes real columns

### PR DESCRIPTION
**Pull request by aaronsteers**
Resolves: #160 

- Remove unused/unneeded `protocol_utils`.
- Rename `CaseInsensitiveDict` class to `StreamRecord`.
- Refactor to all code paths use `StreamRecord` class.
- Add top-level column names are normalized before writing to dict.
- Add handling in SQL processors to normalize names to lower-case before importing from files to SQL table.

### CODE REVIEW

File: airbyte/sources/base.py [hunk 1]
```diff
@@ -73,7 +72,7 @@ def as_temp_files(files_contents: list[dict | str]) -> Generator[list[str], Any,
     finally:
         for temp_file in temp_files:
             with suppress(Exception):
-                temp_file.unlink()
+                Path(temp_file.name).unlink()
```
Comment (aaronsteers) [inline airbyte/sources/base.py:75]: Drive-by fix. `temp_file` doesn't have `unlink()` method. Was being suppressed.

**Review by bindipankhudi — APPROVED** [review]
Looks great!
